### PR TITLE
Fixes #13554 - Setting releated to taxonomies shouldn't be shown if they're disabled

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -48,8 +48,16 @@ def validate(record)
   after_find :readonly_when_overridden_in_SETTINGS
   default_scope -> { order(:name) }
 
-  # The DB may contain settings from disabled plugins - filter them out here
-  scope :live_descendants, -> { where(:category => self.descendants.map(&:to_s)) unless Rails.env.development? }
+  # Filer out settings from disabled plugins and ones releated to taxonomies when required
```
Comment (dLobatog) [inline app/models/setting.rb:51]: @sean797 You can use `Taxonomy.enabled_taxonomies` to find out which ones are enabled and build the query consequently. This is also missing some tests if you can add them? Thank you!

**Comment by sean797** [comment]
[test]

**Comment by sean797** [comment]
Currently this still allows `GET /api/settings/:id` and `PUT /api/settings/:id` even if those settings are relevant, but they don't show on `GET /api/settings`. I will add tests after assuming you're happy with this?

**Comment by dLobatog** [comment]
@sean797 That's alright, thanks!

**Comment by dLobatog** [comment]
@sean797 Notice a few tests are broken - http://ci.theforeman.org/job/test_develop_pr_core/8032/#showFailuresLink please fix and update the PR

**Comment by sean797** [comment]
Thanks @dLobatog. Should be good to merge now if you're happy?

**Comment by dLobatog** [comment]
Merged as 6067d906ff50338a034c60679384d484890e5bda, thanks @sean797!

**Pull request closed without merging**